### PR TITLE
Aiding the Food/Drink Refactor: Bottle Edition

### LINKED ISSUE #8405: More food refactor invisible sprites
Caused by #8285

Noticed with Grenadine
grenadinebottle_empty wasn't added in the PR, so it turns invisible when empty. Discovered 3/8/20

This is going to happen with every bottle in the /obj/item/reagent_containers/food/drinks/bottle/ path if there isn't an iconstate_empty sprite since it's in the parent path's update_icon() that if the bottle has no reagents, the iconstate becomes "iconstate_empty." If there isn't one, it's invisible.

For example I already don't see any "_empty" states for space up, cola, or mountain wind bottles added in the PR so those will probably go invisible, too. Same with like every juice carton, since those are in that path.

Would fix it but I assume the PR author would like to add the missing empty sprites instead, and I'm not a spriter.

### PR DESCRIPTION
**Pull request by Doxxmedearly**
The food/drink refactor gave us empty sprites for bottles and organized it for the future, which is great. It just needed some tweaks.

Added two flags for bottles: NO_EMPTY_STATE, which indicates that there is NOT an empty bottle sprite, and BOTTLE_IS_GLASS, to replace the isGlass var. BOTTLE_IS_GLASS is set as default since most of the bottles fall under it. 

Bottles will display their default icon state when empty (like before) if they're flagged as NO_EMPTY_STATE. This helps for adding future bottles, AND makes it easy to add empty_states to existing ones; you just have to make sure to remove the flag.

If there's better way to do these flags let me know. 

Sprite change is just renaming alco-empty to alco-clear_empty to fit with the format. 

Added /carton to the bottle path to make it easier to differentiate between bottles and juice cartons. 

Also fixed a typo I found which caused tonic water cans not to initialize.

Fixes #8405

### CODE REVIEW

**Comment by Doxxmedearly** [comment]
!review

**Comment by Doxxmedearly** [comment]
reversed the flags. makes way more sense to remove a flag if a state is added than to add one.

**Comment by Doxxmedearly** [comment]
I have no idea what Travis is screaming about

File: code/modules/reagents/reagent_containers/food/drinks/bottle.dm [hunk 1]
```diff
@@ -437,79 +450,66 @@
 
 //////////////////////////JUICES AND STUFF ///////////////////////
 
-/obj/item/reagent_containers/food/drinks/bottle/orangejuice
+/obj/item/reagent_containers/food/drinks/bottle/carton/
+	name = "Generic Carton"
```
Comment (NonQueueingMatt) [inline code/modules/reagents/reagent_containers/food/drinks/bottle.dm]:
```suggestion
	name = "carton"
```

File: code/modules/reagents/reagent_containers/food/drinks/bottle.dm [hunk 2]
```diff
@@ -437,79 +450,66 @@
 
 //////////////////////////JUICES AND STUFF ///////////////////////
 
-/obj/item/reagent_containers/food/drinks/bottle/orangejuice
+/obj/item/reagent_containers/food/drinks/bottle/carton/
```
Comment (NonQueueingMatt) [inline code/modules/reagents/reagent_containers/food/drinks/bottle.dm]:
```suggestion
/obj/item/reagent_containers/food/drinks/bottle/carton
```

**Comment by Doxxmedearly** [comment]
Real contributor hours. Thanks matt

**Pull request merged**